extension/source/types/Rpc.ts

```diff
@@ -31,6 +31,36 @@ export const SendTransactionParams = io.type({
 
 export type SendTransactionParams = io.TypeOf<typeof SendTransactionParams>;
 
+export const TransactionStatus = io.union([
+  io.literal('new'),
+  io.literal('approved'),
+  io.literal('rejected'),
+  io.literal('cancelled'),
+  io.literal('confirmed'),
+  io.literal('failed'),
+]);
+
+export type TransactionStatus = io.TypeOf<typeof TransactionStatus>;
+
+export const QuillTransaction = io.type({
+  id: io.string,
+  chainId: io.string,
+  from: io.string,
+  status: TransactionStatus,
```

Comment:
This was `io.string` before. Using `TransactionStatus` is important to get the most out of TypeScript's checking and make things fit together nicely. This is also a reason why I replaced the enum with a union because enums don't translate as nicely into io types.